### PR TITLE
Refactor verify

### PR DESCRIPTION
**Pull request by rbehjati**
Refactor verify to be able to more easily support different types of provenance verification logic.

### CODE REVIEW

**Review by mariaschett — APPROVED** [review]
LGTM

File: verify/verify.go [hunk 1]
```diff
@@ -24,8 +24,25 @@ import (
 	"github.com/project-oak/transparent-release/slsa"
 )
 
-// Verify verifies a given SLSA provenance file.
-func Verify(provenanceFilePath, gitRootDir string) error {
+// ProvenanceVerifier defines an interface with a single method `Verify` for
+// verifying provenances.
+type ProvenanceVerifier interface {
+	// Verify verifies an Amber/SLSA provenance file in the given path.
```
Comment (tiziano88) [inline verify/verify.go:30]: Is the expectation that a nil error means the verification is successful?

**Review by tiziano88 — APPROVED** [review]
Nice

**Review by rbehjati — COMMENTED** [review]
Thanks :)

**Pull request merged**